src/wp-includes/class-wp-oembed.php

```diff
@@ -539,6 +539,12 @@ public function fetch( $provider, $url, $args = '' ) {
 			if ( is_wp_error( $result ) && 'not-implemented' === $result->get_error_code() ) {
 				continue;
 			}
+			// Use youtube-nocookie.com for YouTube embeds.
+			if ( strpos( $provider, 'https://www.youtube.com/oembed' ) === 0 && ! empty( $result->html ) ) {
+				$from = 'src="https://www.youtube.com/embed/';
+				$to   = 'src="https://www.youtube-nocookie.com/embed/';
+				$result->html = str_replace( $from, $to, $result->html );
+			}
```

Comment:
What about `wp_embed_handler_youtube`?

Reply:
No change required, unless we want to accept youtube-nocookie links as input. I can try to make a separate pull request for that based on jepperask's code.